src/v2i-hub/MapPlugin/src/MapPlugin.h

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+#ifndef MAPPLUGIN_H_
+#define MAPPLUGIN_H_
+
+#include <atomic>
+#include <iostream>
+#include <map>
+#include <memory>
+#include <mutex>
+#include <sstream>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <thread>
+#include <time.h>
+#include <sys/time.h>
+#include <string>
+
+#include <tmx/tmx.h>
+#include <tmx/IvpPlugin.h>
+#include <tmx/messages/IvpBattelleDsrc.h>
+#include <tmx/messages/IvpSignalControllerStatus.h>
+#include <tmx/messages/IvpJ2735.h>
+#include <tmx/j2735_messages/J2735MessageFactory.hpp>
+#include "XmlMapParser.h"
+#include "ConvertToJ2735r41.h"
+#include "inputs/isd/ISDToJ2735r41.h"
+
+#define USE_STD_CHRONO
+#include <FrequencyThrottle.h>
+#include <PluginClientClockAware.h>
+
+#include "utils/common.h"
+#include "utils/map.h"
+
+#include <MapSupport.h>
+
+using namespace std;
+using namespace tmx;
+using namespace tmx::messages;
+using namespace tmx::utils;
+
+namespace MapPlugin {
+
+#if SAEJ2735_SPEC < 63
+UPERframe _uperFrameMessage;
+#endif
+
+class MapFile: public tmx::message {
+public:
+	MapFile(): tmx::message() {}
+	virtual ~MapFile() {}
+
+	std_attribute(this->msg, int, Action, -1, );
```

Comment:
What is the rest of actions this MAP support? Which default action is this initialized to?

Reply:
Default is no action existing. It looks like only action is for finding the file path.